sdks/java/io/jms/src/test/java/org/apache/beam/sdk/io/jms/JmsIOTest.java

```diff
@@ -92,70 +87,49 @@
 import org.apache.beam.sdk.transforms.SerializableBiFunction;
 import org.apache.beam.sdk.values.PCollection;
 import org.apache.beam.vendor.guava.v26_0_jre.com.google.common.base.Throwables;
+import org.apache.qpid.jms.JmsAcknowledgeCallback;
+import org.apache.qpid.jms.JmsConnectionFactory;
+import org.apache.qpid.jms.message.JmsTextMessage;
 import org.joda.time.Duration;
 import org.junit.After;
 import org.junit.Before;
 import org.junit.Rule;
 import org.junit.Test;
 import org.junit.runner.RunWith;
-import org.junit.runners.JUnit4;
+import org.junit.runners.Parameterized;
 import org.mockito.ArgumentCaptor;
 import org.mockito.Mockito;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /** Tests of {@link JmsIO}. */
-@RunWith(JUnit4.class)
+@RunWith(Parameterized.class)
 @SuppressWarnings({
   "rawtypes", // TODO(https://github.com/apache/beam/issues/20447)
 })
-public class JmsIOTest {
-
-  private static final String BROKER_URL = "vm://localhost";
-
-  private static final String USERNAME = "test_user";
-  private static final String PASSWORD = "test_password";
-  private static final String QUEUE = "test_queue";
-  private static final String TOPIC = "test_topic";
-
-  private BrokerService broker;
-  private ConnectionFactory connectionFactory;
-  private ConnectionFactory connectionFactoryWithSyncAcksAndWithoutPrefetch;
-
-  @Rule public final transient TestPipeline pipeline = TestPipeline.create();
+public class JmsIOTest extends CommonJms {
```

Comment:
Can we make JmsIOTest and JmsIOIT own a CommonJms instance to handle connection related things instead of extend it? There is no polymorphism usage involved here and inheritance can make the test class implementation complicated imo.